src/rgw/rgw_op.cc

```diff
@@ -3910,10 +3931,17 @@ void RGWPutObj::execute()
       break;
     }
 
+    /* XXX likely this forces a rebuild of data--should implement a
+     * segmented Digest::Update */
```

Comment:
good catch! i see there's a new templated `ceph::crypto::digest<Digest>(const ceph::buffer::list& bl)` helper function in ceph_crypto.h that calls Update() piecewise. maybe we could split that logic (without the init/finalize) into another helper like this?
```c++
template <typename DigestT>
void digest_update(DigestT& gen, const ceph::buffer::list& bl)
{
  for (auto& p : bl.buffers()) {
    gen.Update((const unsigned char *)p.c_str(), p.length());
  }
}
```

Reply:
cool, didn't know it existed, will update to use it